### PR TITLE
Update histogram tool to use python3 and conda requirements

### PR DESCRIPTION
**Pull request by cat-bro**
histogram has python errors and we haven't been able to run using the legacy tool dependencies.  This updates the tool to use conda installable dependencies and changes in the python script to preserve the original functionality with the updated packages and python3.

Addresses https://github.com/galaxyproject/tools-devteam/issues/554

histogram is identical to another devteam tool dna_filtering.  This could be updated in the same way or deprecated.

### CODE REVIEW

**Comment by cat-bro** [comment]
The test now uses sim_size comparison between expected and output PDFs since direct diff comparison will fail.

Expected output:


Test output:

File: tools/histogram/histogram.py [hunk 1]
```diff
@@ -1,101 +1,108 @@
...
     in_fname = sys.argv[1]
-    out_fname = sys.argv[2] 
+    out_fname = sys.argv[2]
     try:
-        column = int( sys.argv[3] ) - 1
-    except:
-        stop_err( "Column not specified, your query does not contain a column of numerical data." )
+        column = int(sys.argv[3]) - 1
+    except Exception:
+        stop_err("Column not specified, your query does not contain a column of numerical data.")
```
Comment (nsoranzo) [inline tools/histogram/histogram.py:25]:
```suggestion
        sys.exit("Column not specified, your query does not contain a column of numerical data.")
```

File: tools/histogram/histogram.py [hunk 2]
```diff
@@ -1,101 +1,108 @@
...
+from rpy2.robjects import r, vectors
+from rpy2.robjects.packages import importr
+
+
+assert sys.version_info[:2] >= (2, 4)
+
 
 def stop_err(msg):
     sys.stderr.write(msg)
     sys.exit()
```
Comment (nsoranzo) [inline tools/histogram/histogram.py:15]:
```suggestion
```

File: tools/histogram/histogram.py [hunk 3]
```diff
@@ -1,101 +1,108 @@
...
-                r.lines( density )
-            r.dev_off()
-        except Exception, exc:
-            stop_err( "%s" %str( exc ) )
+                    scale_factor = len(matrix) * (histogram['mids'][1] - histogram['mids'][0])  # uniform bandwidth taken from first 2 midpoints
+                    density['y'] = map(lambda x: x * scale_factor, density['y'])
+                graphics.lines(density)
+            grdevices.dev_off()
+        except Exception as exc:
+            stop_err("%s" % str(exc))
```
Comment (nsoranzo) [inline tools/histogram/histogram.py:95]:
```suggestion
            sys.exit("%s" % str(exc))
```

File: tools/histogram/histogram.py [hunk 4]
```diff
@@ -1,101 +1,108 @@
...
     else:
         if i == 0:
             stop_err("Input dataset is empty.")
```
Comment (nsoranzo) [inline tools/histogram/histogram.py:98]:
```suggestion
            sys.exit("Input dataset is empty.")
```

File: tools/histogram/histogram.py [hunk 5]
```diff
@@ -1,101 +1,108 @@
...
         else:
-            stop_err( "All values in column %s are non-numeric." %sys.argv[3] )
+            stop_err("All values in column %s are non-numeric." % sys.argv[3])
```
Comment (nsoranzo) [inline tools/histogram/histogram.py:100]:
```suggestion
            sys.exit("All values in column %s are non-numeric." % sys.argv[3])
```

File: tools/histogram/histogram2.xml [hunk 6]
```diff
@@ -1,8 +1,8 @@
 <tool id="histogram_rpy" name="Histogram" version="1.0.3">
```
Comment (nsoranzo) [inline tools/histogram/histogram2.xml:1]:
```suggestion
<tool id="histogram_rpy" name="Histogram" version="1.0.4">

```

**Comment by cat-bro** [comment]
Thanks @nsoranzo and @bgruening!

**Comment by bgruening** [comment]
Merci!

**Comment by cat-bro** [comment]
This one is merged but the updated histogram tool is not yet available in the shed.  My motivation for updating it has been that Galaxy Australia would like to be able to teach the maxquant tutorial (histogram has recently been added to this).  I could make another PR to address the code quality if this would help.

**Comment by bgruening** [comment]
I restarted a run ... hopefully, this will succeed. Thanks for reporting @cat-bro

**Comment by cat-bro** [comment]
Thank you @bgruening it is in the toolshed now.

**Pull request merged**